### PR TITLE
fix: Bugfixes for how registry is loaded

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sdk/python/feast/registry.py [hunk 1]
```diff
@@ -689,6 +758,16 @@ def get_user_metadata(
     cached_registry_proto_created: Optional[datetime] = None
     cached_registry_proto_ttl: timedelta
 
+    def __new__(
```
Comment (felixwang9817) [inline sdk/python/feast/registry.py:761]: it's not clear to me why this is necessary - can you explain in the PR description what the bug is / how it's being fixed?

also are there any tests that would catch the bugs?

**Review by felixwang9817 — COMMENTED** [review]
just one nit

File: sdk/python/feast/registry.py (hunk 1 above)
Comment (achals) [reply sdk/python/feast/registry.py:761]: https://docs.python.org/3/reference/datamodel.html#object.__new__

Basically, `__new__` is called before `__init__`, and can be used to *change* the object that's being created by a call to a class constructor. So in our case, we are calling `Registry()`, but we want to return a `SqlRegistry` based on the config information. This implementation achieves that.

I can add a test for this specifically.

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**